h35/src/components/Search.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+
+export default function Search({ data }) {
+  const [ name, setName ] = useState(null); //Store the name to search for
+  const [ cards, setCards ] = useState([]); //Store the cards to display
+
+  const imageURL = 'https://thronesapi.com/assets/images/'; //Image URL string
+  
+  //Issue:  1st time name is searched for, nothing happens - have to click Submit 2x; why?
+  const onSubmit = (event) => {
+    //Get and set the name to search for
+    event.preventDefault();
+    const results = [];
+    const target = event.target.charName.value;
+    setName(target);
```

Comment:
setName ís asynchronous which means you can't rely on the value of name to be set yet due to how react handles state updates. If you use target instead of name here it should work fine.

See the "State Updates May Be Asynchronous" section. I would have referenced the new docs, but didn't see this bit in there.
https://legacy.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
Resolved this - thank you!